### PR TITLE
[Filter not working per user timezone]

### PR DESCRIPTION
**Pull request by rmu-odoo**
Hello,

With this PR, Filter "Today" shows all of the timesheet belong to the same day , before the UTC's midnight (at 11AM, local time), if the entry was made it will appear to the previous day. they should not appears on the previous day.

Thanks.

### CODE REVIEW

File: addons/hr_attendance/hr_attendance.py [hunk 1]
```diff
@@ -48,7 +48,8 @@ class hr_attendance(osv.osv):
     def _day_compute(self, cr, uid, ids, fieldnames, args, context=None):
         res = dict.fromkeys(ids, '')
         for obj in self.browse(cr, uid, ids, context=context):
-            res[obj.id] = time.strftime('%Y-%m-%d', time.strptime(obj.name, '%Y-%m-%d %H:%M:%S'))
+            date_start = fields.datetime.context_timestamp(cr, uid, datetime.strptime(obj.name, '%Y-%m-%d %H:%M:%S'), context=context)
+            res[obj.id] = date_start.strftime('%Y-%m-%d')
```
Comment (msh-odoo) [inline addons/hr_attendance/hr_attendance.py]: May be you can use context_today only, something like res[obj.id] = fields.date.context_today(timestamp)
context_today already returns strftime value fit for the date field

**Comment by mart-e** [comment]
Hello,

Sorry but I don't think that will work. `day` is a stored computed field. It will be stored globally, at the first access and return the same value, even if somebody in another timezone access it.

**Comment by rha-odoo** [comment]
actual problem is in the filter where it passed current_date, which is current client's date, where field name will be stored as UTC in db, that's the reason filter doesn't work.

**Comment by mart-e** [comment]
@rha-odoo yes, storing data that's dependent of the user timezone will not work.
Also the field day is no longer present in 8.0 so I would prefer a patch compatible for the next version.
Thanks

**Comment by mart-e** [comment]
Closing as not fixing for 7.0. In 7.0 we had not the valid method to properly handle timezones in the web client. It was fixed at 5b7ade9 (present in 8.0) but it's a change of behaviour so can not be backported.

**Pull request closed without merging**